v0/destinations/revenue_cat/transform.js

```diff
@@ -0,0 +1,142 @@
+const set = require("set-value");
+const { defaultRequestConfig } = require("rudder-transformer-cdk/build/utils");
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  constructPayload,
+  ErrorMessage,
+  defaultPostRequestConfig,
+  extractCustomFields,
+  defaultGetRequestConfig
+} = require("../../util");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  REVENUE_CAT_IDENTIFY_EXCLUSION
+} = require("./config");
+
+const trackResponseBuilder = async (message, category, { Config }) => {
+  if (!Config.xPlatform) {
+    throw new CustomError("[REVENUE CAT] X-Platform is required field.", 400);
+  }
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  // che
+  if (
+    message.properties.products &&
+    Array.isArray(message.properties.products)
+  ) {
+    const responseArray = [];
+    const productList = message.properties.products;
```

Comment:
Better to store the `message.properties.products` outside loop in a variable rather than doing dot notion multiple times

Reply:
Yeah, Done